static/compiler.js

```diff
@@ -405,30 +580,57 @@ function Compiler(domRoot, origFilters, windowLocalPrefix, onChangeCallback, lan
         }
         return compiler;
     }
-
-    function setCompilers(compilers, defaultCompiler) {
-        domRoot.find('.compilers li').remove();
+    
+    // added has auxiliary to setCompilers, in order not to break interface
+    // TODO : consider refactoring as some tasks are repeated
+    function setCompilersInSlot(compilers, defaultCompiler, slot) {
+        console.log("[INIT] in setCompilersInSlot(), compilers = "+
+        JSON.stringify(compilers)+", slot = "+slot.id);
+        domRoot.find('#slot'+slot.id+' .compilers li').remove();
         compilersById = {};
         compilersByAlias = {};
+        // fills the compiler list
         $.each(compilers, function (index, arg) {
             compilersById[arg.id] = arg;
             if (arg.alias) compilersByAlias[arg.alias] = arg;
+            // li : list element
             var elem = $('<li><a href="#">' + arg.name + '</a></li>');
-            domRoot.find('.compilers').append(elem);
+            domRoot.find('#slot'+slot.id+' .compilers').append(elem);
```

Comment:
maybe consider a function in the `slot` to return its domRoot. Then all this stuff with `'#slot'+slot.id+' ....'` could be `slot.node.find(".compilers")` etc